### PR TITLE
Add metrics pipeline with carbon2 format

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Depends on #3

### CODE REVIEW

File: exporter/sumologicexporter/sender.go [hunk 1]
```diff
@@ -267,3 +279,132 @@ func (s *sender) batch(log pdata.LogRecord, metadata Fields) ([]pdata.LogRecord,
...
+		switch record.metric.DataType() {
+		case pdata.MetricDataTypeIntGauge:
+			for i := 0; i < record.metric.IntGauge().DataPoints().Len(); i++ {
+				dataPoint := record.metric.IntGauge().DataPoints().At(i)
+				nextLines = append(nextLines,
+					fmt.Sprintf("%s  %d %d",
+						s.filter.Carbon2TagString(record),
+						dataPoint.Value(),
+						dataPoint.Timestamp()/1e9,
+					),
```
Comment (pmalek-sumo) [inline exporter/sumologicexporter/sender.go:333]: I believe I was suggesting to extract those `Sprintf`s

Comment (sumo-drosiek) [reply exporter/sumologicexporter/sender.go:333]: yup, I'm working on that meantime

Comment (sumo-drosiek) [reply exporter/sumologicexporter/sender.go]: done 🤷

Comment (sumo-drosiek) [reply exporter/sumologicexporter/sender.go]: It was the last thing from previous opened PR

File: exporter/sumologicexporter/exporter.go [hunk 2]
```diff
@@ -178,3 +178,101 @@ func (se *sumologicexporter) pushLogsData(_ context.Context, ld pdata.Logs) (int
...
+		exporterhelper.WithTimeout(exporterhelper.TimeoutSettings{Timeout: 0}),
+		exporterhelper.WithRetry(cfg.RetrySettings),
+		exporterhelper.WithQueue(cfg.QueueSettings),
+	)
+}
+
+// pushMetricsData groups data with common metadata and send them as separate batched requests
+// it returns number of unsent metrics and error which contains list of dropped records
+// so they can be handle by the OTC retry mechanism
+func (se *sumologicexporter) pushMetricsData(_ context.Context, ld pdata.Metrics) (int, error) {
```
Comment (pmalek-sumo) [inline exporter/sumologicexporter/exporter.go:206]: Can we mark those unused contexts with TODO comments?
Otelcol community is fine with such an approach?

Comment (sumo-drosiek) [reply exporter/sumologicexporter/exporter.go:206]: What do you want to put into those TODO?

Comment (pmalek-sumo) [reply exporter/sumologicexporter/exporter.go:206]: > contextify this method

:) Ideally this should be used when issuing requests. Is it done this way in other exporters?

File: exporter/sumologicexporter/exporter.go [hunk 3]
```diff
@@ -178,3 +178,101 @@ func (se *sumologicexporter) pushLogsData(_ context.Context, ld pdata.Logs) (int
...
+		resource := rms.At(i)
+
+		if resource.IsNil() {
+			continue
+		}
+
+		attributes = resource.Resource().Attributes()
+
+		// iterate over InstrumentationLibraryMetrics
+		ilm := resource.InstrumentationLibraryMetrics()
```
Comment (pmm-sumo) [inline exporter/sumologicexporter/exporter.go:226]: This is actually a slice, so frequently it's associated with `ilms` variable (not that it matters that much)

File: exporter/sumologicexporter/exporter.go [hunk 4]
```diff
@@ -178,3 +178,101 @@ func (se *sumologicexporter) pushLogsData(_ context.Context, ld pdata.Logs) (int
...
+		for j := 0; j < ilm.Len(); j++ {
+			library := ilm.At(j)
```
Comment (pmm-sumo) [inline exporter/sumologicexporter/exporter.go:228]: This is frequently called `ilm` and it keeps both information about the library and collection of records, so using `library` is not a good fit here I think

Would we want to push forward information about the InstrumentationLibrary itself? (name, version)? It's present in `InstrumentationLibrary()`

File: exporter/sumologicexporter/exporter.go (hunk 2 above)
Comment (sumo-drosiek) [reply exporter/sumologicexporter/exporter.go:206]: It's common approach across the exporters

File: exporter/sumologicexporter/exporter.go (hunk 4 above)
Comment (sumo-drosiek) [reply exporter/sumologicexporter/exporter.go:228]: I will rename varabiables 👍 

Maybe add those information about `InstrumentationLibrary` later as separate PR

File: exporter/sumologicexporter/exporter.go (hunk 2 above)
Comment (sumo-drosiek) [reply exporter/sumologicexporter/exporter.go:206]: but seems that we will have to pass it to the sender anyway

File: exporter/sumologicexporter/sender_test.go [hunk 5]
```diff
@@ -376,3 +376,135 @@ func TestBuffer(t *testing.T) {
...
+	metrics := exampleTwoIntMetrics()
+
+	dropped, err := test.s.batchMetric(metrics[0])
+	require.NoError(t, err)
+	assert.Nil(t, dropped)
+	assert.Equal(t, 1, test.s.countMetrics())
+	assert.Equal(t, metrics[:1], test.s.metricBuffer)
+
+	dropped, err = test.s.batchMetric(metrics[1])
+	require.Equal(t, err, nil)
```
Comment (pmalek-sumo) [inline exporter/sumologicexporter/sender_test.go:394]:
```suggestion
	require.NoError(t, err)
```

**Comment by sumo-drosiek** [comment]
Closing in favor of #12 and #10

**Pull request closed without merging**